TestFramework.cpp

```diff
@@ -0,0 +1,408 @@
+/*
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author Michael Ray
+ * @since 3/22/2017
+ * @version 1.0.0
+ * 
+ */
+#include "TestFramework.hpp"
+
+/*/////////////////////////////////////////////////////////////////////////////
+//								Initialization								 //
+/////////////////////////////////////////////////////////////////////////////*/
+
+/**
+  * Constructor for the test framework. Add all of the PinMaps to the pinout, as well as CI test shield pins
+**/
+TestFramework::TestFramework() {
+	map_pins(PinMap_ADC, AnalogInput);
+	map_pins(PinMap_DAC, AnalogOutput);
+	map_pins(PinMap_PWM, DigitalIO);
+	map_pins(PinMap_ADC, DigitalIO);
+	// map_pins(PinMap_DAC, DigitalIO);
+	map_pins(PinMap_I2C_SDA, DigitalIO);
+	map_pins(PinMap_I2C_SCL, DigitalIO);
+	map_pins(PinMap_SPI_SCLK, DigitalIO);
+	map_pins(PinMap_SPI_MOSI, DigitalIO);
+	// map_pins(PinMap_SPI_MISO, DigitalIO);
+	map_pins(PinMap_SPI_SSEL, DigitalIO);
+	pinout[BusIO] = pinout[DigitalIO];
+	map_pins(PinMap_PWM, PWM);
+	map_pins(PinMap_I2C_SDA, I2C_SDA);
+	map_pins(PinMap_I2C_SCL, I2C_SCL);
+	map_pins(PinMap_SPI_SCLK, SPI_CLK);
+	map_pins(PinMap_SPI_MOSI, SPI_MOSI);
+	map_pins(PinMap_SPI_MISO, SPI_MISO);
+	map_pins(PinMap_SPI_SSEL, SPI_CS);
+	setup_cits_pins();
+}
+
+/**
+  * Add all of the CI Test Shield pins to the pinout vector
+**/
+void TestFramework::setup_cits_pins() {
+	pinout[CITS_AnalogInput].push_back(construct_pinmap(MBED_CONF_APP_AIN_0));
+	pinout[CITS_AnalogInput].push_back(construct_pinmap(MBED_CONF_APP_AIN_1));
+	pinout[CITS_AnalogInput].push_back(construct_pinmap(MBED_CONF_APP_AIN_2));
+	pinout[CITS_AnalogInput].push_back(construct_pinmap(MBED_CONF_APP_AIN_3));
+	pinout[CITS_AnalogInput].push_back(construct_pinmap(MBED_CONF_APP_AIN_4));
+	pinout[CITS_AnalogInput].push_back(construct_pinmap(MBED_CONF_APP_AIN_5));
+	pinout[CITS_AnalogOutput].push_back(construct_pinmap(MBED_CONF_APP_AOUT));
+	pinout[CITS_DigitalIO].push_back(construct_pinmap(MBED_CONF_APP_DIO_0));
```

Comment:
Valid point. The D0/D1 lines of code don't actually do anything anyway because when I dynamically find digital pins to test, the D0/D1 pins aren't pulled into the list because they are associated with UART and not Digital or PWM pinmaps. Therefore when the dynamic pin list is cross referenced with the CI Test Shield pins, the MBED_CONF_APP_DIO0/1 pins won't be used. I will remove lines 64 and 65 anyway to remove confusion